### ISSUE TITLE
[Bug]: Docker M1 Mac fails to open new page (`Target.createTarget`)

### ISSUE THREAD

**Issue by davidbielik** [puppeteer/puppeteer#7916]
### Bug description

```typescript
const puppeteer = require('puppeteer');
(async function () {
  const browser = await puppeteer.launch({
    headless: true,
    args: [
      '--disable-gpu', // if i remove this, strangely the process exits unexpectedly
      '--disable-dev-shm-usage',
      '--no-sandbox',
      '--disable-setuid-sandbox',
    ],
  });
  await browser.newPage();
})();
```

**Dockerfile**

```
FROM --platform=linux/amd64 node:16.13.1-slim

RUN apt-get update && \
    apt-get -y install vim xvfb gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 \
      libdbus-1-3 libexpat1 libfontconfig1 libgbm1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 \
      libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 \
      libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 \
      libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget python make g++

RUN rm -rf /var/lib/apt/lists/*

# todo: automate this version
RUN npm install --save puppeteer@13.5.1

COPY package*.json ./
RUN npm ci;
RUN mkdir -p /home/node/app && cp -a /tmp/node_modules /home/node/app/

RUN groupadd -r pptruser && useradd -r -g pptruser -G audio,video pptruser \
  && mkdir -p /home/pptruser/Downloads \
  && chown -R pptruser:pptruser /home/pptruser \
  && chown -R pptruser:pptruser /home/node/app

RUN ./node_modules/typescript/bin/tsc -b --verbose

USER pptruser
```

### Puppeteer version

13.5.1

### Node.js version

16.13.1

### npm version

8.3.0

### What operating system are you seeing the problem on?

macOS (M1 Mac)

### Relevant log output

```shell
ProtocolError: Protocol error (Target.createTarget): Target closed.
    at /home/node/app/node_modules/puppeteer/lib/cjs/puppeteer/common/Connection.js:75:24
    at new Promise (<anonymous>)
    at Connection.send (/home/node/app/node_modules/puppeteer/lib/cjs/puppeteer/common/Connection.js:71:16)
    at Browser._createPageInContext (/home/node/app/node_modules/puppeteer/lib/cjs/puppeteer/common/Browser.js:261:53)
    at BrowserContext.newPage (/home/node/app/node_modules/puppeteer/lib/cjs/puppeteer/common/Browser.js:520:30)
    at Browser.newPage (/home/node/app/node_modules/puppeteer/lib/cjs/puppeteer/common/Browser.js:254:37)
    at /home/node/app/src/test.js:17:19
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by schlomo** [comment]
I also have this `Target closed` problem on M1 Macbook. Native (arm64) Node `v17.5.0` and puppeteer `v13.4.1`.

Happens at the 3rd `browser.newPage()` call.

**Comment by artyil** [comment]
I was hoping that `PUPPETEER_EXPERIMENTAL_CHROMIUM_MAC_ARM` will fix that as well, but it didn't (at least for me). I still can't open pages in pptr docker on my m1 macbook.

I have the following error:
```
ProtocolError: Protocol error (Target.createTarget): Target closed.
    at /src/node_modules/puppeteer/lib/cjs/puppeteer/common/Connection.js:75:24
    at new Promise (<anonymous>)
    at Connection.send (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/Connection.js:71:16)
    at Browser._createPageInContext (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/Browser.js:287:53)
    at BrowserContext.newPage (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/Browser.js:566:30)
    at Browser.newPage (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/Browser.js:280:37
```

Are there any plans to address this issue soon ⛈️  ?

**Comment by sepatel** [comment]
> I was hoping that `PUPPETEER_EXPERIMENTAL_CHROMIUM_MAC_ARM` will fix that as well, but it didn't (at least for me). I still can't open pages in pptr docker on my m1 macbook.
> 
> I have the following error:
> 
> ```
> ProtocolError: Protocol error (Target.createTarget): Target closed.
>     at /src/node_modules/puppeteer/lib/cjs/puppeteer/common/Connection.js:75:24
>     at new Promise (<anonymous>)
>     at Connection.send (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/Connection.js:71:16)
>     at Browser._createPageInContext (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/Browser.js:287:53)
>     at BrowserContext.newPage (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/Browser.js:566:30)
>     at Browser.newPage (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/Browser.js:280:37
> ```
> 
> Are there any plans to address this issue soon cloud_with_lightning_and_rain ?

We are running into this issue on our end as well. Running puppeteer inside of docker on an M1 just doesn't seem to work. Fine on an Intel Mac and on Linux but not working on the M1 mac.

We've tried basing it from
FROM --platform=linux/amd64 public.ecr.aws/lambda/nodejs:16
FROM --platform=linux/arm64/v8 public.ecr.aws/lambda/nodejs:16
FROM --platform=linux/amd64 node:16

with the 1st one of those being the one we'd really like working most.

**Comment by artyil** [comment]
Thank @sepatel ! I hope this issue will get more interest from the pptr team and they will investigate and post back. 
@davidbielik can you share if you were able to resolve?

**Comment by andyscraven** [comment]
> We're marking this issue as unconfirmed because it has not had recent activity and we weren't able to confirm it yet. It will be closed if no further activity occurs within the next 30 days.

I am having the same issue.

I am using Laravel Sail as my docker solution and am using a package called Roach for scraping, which in turn uses spatie/browsershot, which in turn uses Puppeteer.

Any help would be much appreciated.

Can I ask how I would force the npm install to use PUPPETEER_EXPERIMENTAL_CHROMIUM_MAC_ARM

I want to set that env variable before I run

`sail npm i puppeteer `

**Comment by harushi99** [comment]
Hello,
I am also having the same issue running on M1. I al ready posted a [stackoverflow](https://stackoverflow.com/questions/73169418/puppeteer-core-on-the-aws-lambda-docker-image-error-target-closed) question with the gihub. But basically running the code on my Mac will fail and running it elsewhere works just fine.

**Comment by andyscraven** [comment]
> Hello, I am also having the same issue running on M1. I already posted a [stackoverflow](https://stackoverflow.com/questions/73169418/puppeteer-core-on-the-aws-lambda-docker-image-error-target-closed) question with the github. But basically running the code on my Mac will fail and running it elsewhere works just fine. What was weird was that few days ago it was working just fine on my Mac too, I am not sure what happened.
> 
> Are there any solutions for this?

I managed to get it installed using PUPPETEER_EXPERIMENTAL_CHROMIUM_MAC_ARM

and now I I can use the ExecuteJavascriptMiddleware Middleware of Roach, which uses Puppeteer but for some reason it doesn't find the installed chrome.

Still trying to solve it!

**Comment by dobreapa** [comment]
I have the same issue running on M1 Pro.
Node: Docker image node:16.15.0
Puppeteer: 14.3.0

**Comment by CeterusMike** [comment]
I can provide a little bit of context here after looking into this for a while yesterday. Basically the issue boils down to Chrome relying on a function called `ptrace`. When you run an amd64 Docker image on an M1 Mac, it runs the image through an emulation layer called `qemu`. Unfortunately, `qemu` does not implement `ptrace`.

So because `ptrace` is not in `qemu`, Chrome crashes. My understanding is that `qemu` is only used when running an amd64 image on an arm64 machine, which I'm assuming most of us having this issue are doing.

In other words, the issue specifically manifests only when using Chrome through `qemu` inside of Docker on an M1 Mac. And since the issue is in `qemu`, I believe it's on them to fix it (it's a very longstanding issue, so not holding my breath), and then that fix needs to make its way into Docker, and then hopefully Chrome will play nicely on M1 Macs using Docker.

I have seen suggestions that running a native arm64 image might be a solution since it wouldn't require `qemu` but I haven't gotten this working yet to know for sure.

**Comment by SteffenAnders** [comment]
Same issues here on Mac M1 with latest puppeteer and browserless/chrome-stable docker container.

**Comment by nick-massie** [comment]
Same issue. Really want to purchase an M1 but waiting in this issue to get fixed. I gained 2 new gray hairs on this issue.

**Comment by montehurd** [comment]
Yep, same issue.

**Comment by LofoWalker** [comment]
I've fixed it by adding --single-process in the launch args just as below : 

```
 let browser = await puppeteer.launch({
            headless: true,
            args: [
                '--no-sandbox',
                '--disable-setuid-sandbox',
                '--disable-dev-shm-usage',
                '--single-process',
                "--disable-web-security",
                '--no-zygote'
            ]
        });
```

**Comment by taletski** [comment]
> I've fixed it by adding --single-process in the launch args just as below :
> 
> ```
>  let browser = await puppeteer.launch({
>             headless: true,
>             args: [
>                 '--no-sandbox',
>                 '--disable-setuid-sandbox',
>                 '--disable-dev-shm-usage',
>                 '--single-process',
>                 "--disable-web-security",
>                 '--no-zygote'
>             ]
>         });
> ```

Could you please share which image did you use?

**Comment by LofoWalker** [comment]
Yeah sure : 

# stable version DO NOT TOUCH or -> [error] https://api.tiles.mapbox.com/mapbox-gl-js/v1.7.0/mapbox-gl.js:30:18711:  JSHandle@error
FROM zenika/alpine-chrome:89-with-node

ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD 1
ENV PUPPETEER_EXECUTABLE_PATH /usr/bin/chromium-browser
ENV NODE_ENV dev
ENV PORT 8080
ENV URL_SUFFIX ""

USER root

WORKDIR /usr/src/app/

COPY . /usr/src/app/

COPY --chown=chrome package.json package-lock.json ./
COPY --chown=chrome . ./

RUN npm install

EXPOSE 8080
CMD ["npm", "start"]

**Comment by joey-i** [comment]
It worked like a charm !!
Thanks @LofoWalker , you saved me wasting another hours to shoot this bug.

**Comment by tuckerbeauchamp** [comment]
Any update on this? I have a Windows 10 machine that runs my app just fine.

As soon as I switch to my M1 Mac I get 2 different errors.

- Error: socket hang up
- Protocol error (Target.setAutoAttach): Target closed.

**Comment by OrKoN** [comment]
will use https://github.com/puppeteer/puppeteer/issues/9763 for tracking
